lib/phoenix/endpoint/supervisor.ex

```diff
@@ -381,25 +390,38 @@ defmodule Phoenix.Endpoint.Supervisor do
   end
 
   defp warmup_static(endpoint) do
-    for {key, value} <- cache_static_manifest(endpoint) do
+    warmup_static(endpoint, cache_static_manifest(endpoint))
+    endpoint.static_path("/")
+  end
+
+  defp warmup_static(endpoint, %{"latest" => latest, "digests" => digests}) do
+    Enum.each(Map.keys(latest), fn key ->
```

Comment:
```suggestion
    Enum.each(latest, fn {key, _} ->
```

To avoid traversing the keys twice.

Reply:
Will fix in next commit.